extensions/podman/packages/extension/src/remote/podman-remote-ssh-tunnel.spec.ts

```diff
@@ -21,25 +21,54 @@ import { type AddressInfo, createConnection, createServer } from 'node:net';
 import { tmpdir } from 'node:os';
 import { join } from 'node:path';
 
+import type { Event, EventEmitter } from '@podman-desktop/api';
+import { Disposable } from '@podman-desktop/api';
+import type { ServerChannel, WindowChangeInfo } from 'ssh2';
 import { Server } from 'ssh2';
 import { generatePrivateKey } from 'sshpk';
-import { beforeAll, beforeEach, expect, test, vi } from 'vitest';
+import { afterEach, beforeAll, beforeEach, describe, expect, test, vi } from 'vitest';
 
 import { PodmanRemoteSshTunnel } from './podman-remote-ssh-tunnel';
 
 // mock the API
-vi.mock('@podman-desktop/api', async () => {
+vi.mock('@podman-desktop/api', () => {
+  /**
+   * Mock the {@link EventEmitter} class logic
+   */
+  class EventEmitterMock<T> implements EventEmitter<T> {
```

Comment:
don't ask me why, vitest is just crashing when I have this outside the vi.mock.

I could do something similar to quadlet, and have this define in `vitest.setup.ts`